### PR TITLE
Skip more than a single leading 0 in exponent display

### PR DESCRIPTION
**Pull request by ddeclerck**

### CODE REVIEW

File: tests/testsuite.src/run_misc.at [hunk 1]
```diff
@@ -15226,3 +15226,26 @@ AT_CHECK([$COMPILE prog.cob])
 AT_CHECK([$COBCRUN_DIRECT ./prog], [0], [], [])
 
 AT_CLEANUP
+
+
+AT_SETUP([DISPLAY FLOAT no leading zeroes in exponent])
```
Comment (GitMensch) [inline tests/testsuite.src/run_misc.at:15231]: just go with "DISPLAY FLOAT" here - and please add keywords FLOAT-SHORT FLOAT-LONG COMP-1 COMP-2 as AT_KEYWORDS

File: tests/testsuite.src/run_misc.at [hunk 2]
```diff
@@ -15226,3 +15226,26 @@ AT_CHECK([$COMPILE prog.cob])
...
+AT_DATA([prog.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01 X  FLOAT-SHORT  VALUE 0.000012345.
+       01 Y  FLOAT-LONG   VALUE 0.000012345.
+       PROCEDURE DIVISION.
+           DISPLAY X.
+           DISPLAY Y WITH NO ADVANCING.
```
Comment (GitMensch) [inline tests/testsuite.src/run_misc.at:15242]: use either `DISPLAY X SPACE Y NO ADVANCING` or just two simple `DISPLAY`s

File: tests/testsuite.src/run_misc.at [hunk 3]
```diff
@@ -15226,3 +15226,26 @@ AT_CHECK([$COMPILE prog.cob])
...
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE prog.cob])
+AT_CHECK([$COBCRUN_DIRECT ./prog], [0],
```
Comment (GitMensch) [inline tests/testsuite.src/run_misc.at:15247]: please add a comment that each COBOL environment do display them in a different way and we check here for internal consistency

File: libcob/termio.c [hunk 4]
```diff
@@ -225,9 +225,12 @@ clean_double (char *wrk)
 	char *pos = strrchr (wrk, 'E');
 
 	if (pos) {
+		char *src;
 		pos += 2; /* skip E+ */
-		if (pos[0] == '0') {
-			memmove (pos, pos + 1, strlen (pos));
+		/* skip leading zeroes */
```
Comment (GitMensch) [inline libcob/termio.c:230]: please add a comment that each COBOL environment do display them in a different way and we ensure here for internal consistency (and if there's enough request we may add different output formats later)

**Review by GitMensch — APPROVED** [review]
code is good to go, please adjust the testcase, add a comment as outlined

**Comment by ddeclerck** [comment]
Merged in SVN @ 5408.

**Pull request closed without merging**